### PR TITLE
Adding blocks, which also define alignment of components

### PR DESCRIPTION
**Pull request by vrnimje**
Adding blocks, which helps user to define alignment of components inside them.
Fixing : #3 
But current implementation raises error with "Exit" function defined for Buttons inside nested blocks

### CODE REVIEW

**Comment by SAtacker** [comment]
@vrnimje please modify the tests as per the new grammar

File: include/quick-ftxui.hpp [hunk 1]
```diff
@@ -128,7 +132,22 @@ struct node_printer : boost::static_visitor<> {
         : indent(indent), data(data_) {}
 
     void operator()(client::quick_ftxui_ast::expression const &expr) const {
-        ast_printer(data, indent + tabsize)(expr);
+        component_meta_data data_nest;
```
Comment (SAtacker) [inline include/quick-ftxui.hpp]: Because screen is uninitialised it is a null pointer and this is a major issue.

**Review by SAtacker — CHANGES_REQUESTED** [review]
```diff
diff --git a/include/quick-ftxui.hpp b/include/quick-ftxui.hpp
index 1c4812d..07ff7ef 100644
--- a/include/quick-ftxui.hpp
+++ b/include/quick-ftxui.hpp
@@ -132,16 +132,18 @@ struct node_printer : boost::static_visitor<> {
         : indent(indent), data(data_) {}
 
     void operator()(client::quick_ftxui_ast::expression const &expr) const {
-        component_meta_data data_nest;
+        component_meta_data data_nest{data->screen};
         ast_printer(&data_nest, indent + tabsize)(expr);
 
         switch (expr.align) {
         case quick_ftxui_ast::block_alignment::VERTICAL: {
-            data->components.push_back(ftxui::Container::Vertical({(data_nest).components}));
+            data->components.push_back(
+                ftxui::Container::Vertical({(data_nest).components}));
             break;
         }
         case quick_ftxui_ast::block_alignment::HORIZONTAL: {
-            data->components.push_back(ftxui::Container::Horizontal({(data_nest).components}));
+            data->components.push_back(
+                ftxui::Container::Horizontal({(data_nest).components}));
             break;
         }
         default:
diff --git a/src/quick-ftxui.cpp b/src/quick-ftxui.cpp
index 65af9cc..4020652 100644
--- a/src/quick-ftxui.cpp
+++ b/src/quick-ftxui.cpp
@@ -48,22 +48,24 @@ int main(int argc, char **argv) {
         printer(expression);
         if (data.components.size()) {
             switch (expression.align) {
-                case client::quick_ftxui_ast::block_alignment::HORIZONTAL: {
-                    auto component =
-                        ftxui::Container::Horizontal(std::move(data.components));
-                    auto main_renderer = ftxui::Renderer(
-                        component, [&] { return ftxui::vbox({component->Render()}); });
-                    screen.Loop(main_renderer);
-                    break;
-                }
-                case client::quick_ftxui_ast::block_alignment::VERTICAL: {
-                    auto component =
-                        ftxui::Container::Vertical(std::move(data.components));
-                    auto main_renderer = ftxui::Renderer(
-                        component, [&] { return ftxui::vbox({component->Render()}); });
-                    screen.Loop(main_renderer);
-                    break;
-                }     
+            case client::quick_ftxui_ast::block_alignment::HORIZONTAL: {
+                auto component =
+                    ftxui::Container::Horizontal(std::move(data.components));
+                auto main_renderer = ftxui::Renderer(component, [&] {
+                    return ftxui::vbox({component->Render()});
+                });
+                screen.Loop(main_renderer);
+                break;
+            }
+            case client::quick_ftxui_ast::block_alignment::VERTICAL: {
+                auto component =
+                    ftxui::Container::Vertical(std::move(data.components));
+                auto main_renderer = ftxui::Renderer(component, [&] {
+                    return ftxui::vbox({component->Render()});
+                });
+                screen.Loop(main_renderer);
+                break;
+            }
             }
         }
     } else {
```

File: include/quick-ftxui.hpp (hunk 1 above)
Comment (vrnimje) [reply include/quick-ftxui.hpp]: Ok, understood. I have made the changes here https://github.com/SAtacker/quick-ftxui/pull/11/commits/3e682ebaf06db1fe527688ffaf9af97dd197ada1

**Comment by SAtacker** [comment]
@vrnimje please fix clang-formatting https://github.com/SAtacker/quick-ftxui/pull/11/checks

**Review by SAtacker — APPROVED** [review]

**Pull request merged**